torch/onnx/symbolic_opset9.py

```diff
@@ -2384,6 +2400,7 @@ def symbolic(g, *args):
 def _dim_arange(g, like, dim):
     like_shape = g.op("Shape", like)
     stop = g.op("Gather", like_shape, g.op("Constant", value_t=torch.tensor(dim)), axis_i=0)
+    # Caffe2-specific op
```

Comment:
Is this (and the opset 11 version of it) and the quantized ops the reason for adding a new operator export type? Can you utilize flag `PYTORCH_ONNX_CAFFE2_BUNDLE` to enable caffe2 fallback only (ensuring caffe2 use cases are the same), such that achieving "strict" aten fallback by default?

Reply:
Mostly, yes. I did consider not adding a new operator export type and use `PYTORCH_ONNX_CAFFE2_BUNDLE` to enable caffe2 specific stuff.

One of the things that drove me off is that even by not setting `-DPYTORCH_ONNX_CAFFE2_BUNDLE`, a couple of caffe2 build flags are still enabled` during pytorch build:

```
//Master flag to build Caffe2
BUILD_CAFFE2:BOOL=ON

//Build Caffe2 operators
BUILD_CAFFE2_OPS:BOOL=ON

//If set, build protobuf inside libcaffe2.so.
CAFFE2_LINK_LOCAL_PROTOBUF:BOOL=ON

CAFFE2_THIRD_PARTY_ROOT:STRING=/path/to/pytorch/third_party

//Test CAFFE2_COMPILER_SUPPORTS_AVX512_EXTENSIONS
CAFFE2_COMPILER_SUPPORTS_AVX512_EXTENSIONS:INTERNAL=1

//Test CAFFE2_EXCEPTION_PTR_SUPPORTED
CAFFE2_EXCEPTION_PTR_SUPPORTED:INTERNAL=1
```

In fact, tests such as `test/onnx/test_pytorch_onnx_caffe2.py` and `test/onnx/test_pytorch_onnx_caffe2_quantized.py` are still runnable even without `PYTORCH_ONNX_CAFFE2_BUNDLE` set.

The fact caffe2 code/tests are available in regular builds and the fact `PYTORCH_ONNX_CAFFE2_BUNDLE` had no comment/document specifying the expected behavior for it made me wary in exploring such flag.

The only use I found for it was defining the default `operator_export_type` when the user didn't use it during export:
```
if not operator_export_type:
    if torch.onnx.PYTORCH_ONNX_CAFFE2_BUNDLE:
        operator_export_type = OperatorExportTypes.ONNX_ATEN_FALLBACK
    else:
        operator_export_type = OperatorExportTypes.ONNX
```

If you feel we can still unify the `operator_export_type` through `PYTORCH_ONNX_CAFFE2_BUNDLE` I am more than happy to give it a shot